src/tests.rs

```diff
@@ -3934,6 +4058,52 @@ mod collator_rewards {
 			);
 		});
 	}
+
+	#[test]
+	fn claim_rewards_other_with_invalid_origin_should_fail() {
+		new_test_ext().execute_with(|| {
+			initialize_to_block(1);
+
+			// Register a candidate
+			register_candidates(4..=4);
+			lock_for_staking(3..=3);
+
+			// Staker 3 stakes on candidate 4.
+			assert_ok!(CollatorStaking::stake(
+				RuntimeOrigin::signed(3),
+				vec![StakeTarget { candidate: 4, stake: 40 }].try_into().unwrap()
+			));
+
+			// Check the collator's counter and staker's checkpoint. Both should be zero, as no
+			// rewards were distributed.
+			assert_eq!(Counters::<Test>::get(&3), FixedU128::zero());
+			assert_eq!(CandidateStake::<Test>::get(&3, &4).checkpoint, FixedU128::zero());
+
+			// Skip session 0, as there are no rewards for this session.
+			initialize_to_block(10);
+
+			// Generate 10 as rewards on the pot generated during session 1.
+			assert_ok!(Balances::mint_into(
+				&CollatorStaking::account_id(),
+				Balances::minimum_balance() + 10
+			));
+
+			// Move to session 2.
+			initialize_to_block(20);
+
+			// Now we have 10 units of rewards being distributed. 20% goes to the collator, and 80%
+			// goes to stakers, so total 8 for stakers. The collator's counter should be the ratio
+			// between the rewards obtained and the total stake deposited in it.
+			assert_eq!(Counters::<Test>::get(&4), FixedU128::from_rational(8, 40));
+
+			// The current checkpoint does not vary, as the staker did not claim the rewards yet.
+			assert_eq!(CandidateStake::<Test>::get(&4, &3).checkpoint, FixedU128::zero());
+
+			// Invalid Origin
+			assert_noop!(CollatorStaking::claim_rewards_other(RuntimeOrigin::root(), 3), BadOrigin);
```

Comment:
To test this functionality you don't need to generate rewards. This test could be composed of a single line, or at least there is no need to check the checkpoints and counter.